risc0/cargo-risczero/build.rs

```diff
@@ -37,7 +37,29 @@ mod runtime {
     }
 }
 
+mod benchmark {
+    pub fn build_benchmark_elf(path: &str) {
+        let benchmark_elf = std::env::current_dir().unwrap().join(path);
+        if let Ok(content) = std::fs::read(&benchmark_elf) {
+            std::fs::write(&benchmark_elf, content).unwrap();
+        } else {
+            build_elf(benchmark_elf.to_str().unwrap());
+        }
+    }
+
+    fn build_elf(path: &str) {
+        let is_success = std::process::Command::new("cargo")
+            .args(["xtask", "gen-benchmark", "--path", path])
+            .status()
+            .unwrap()
+            .success();
+        assert!(is_success);
+    }
+}
+
```

Comment:
I think I might have found a way to compile it automatically during build time! 😄

Reply:
Giving credit where credit is due, I found it already in our code base :)